crates/ubrn_bindgen/src/bindings/react_native/gen_typescript/templates/ErrorTemplate.ts

```diff
@@ -48,20 +49,23 @@ const {{ ffi_converter_name }} = (() => {
         }
         write(value: TypeName, into: RustBuffer): void {
             {%- if e.is_flat() %}
-
+            switch ((value as any)._uniffiVariantName) {
             {%- for variant in e.variants() %}
-            if (value instanceof {{ type_name }}.{{ variant.name()|class_name(ci) }}) {
-                intConverter.write({{ loop.index0 }}, into);
-            } else
-            {%- endfor %} {
-                throw new UniffiInternalError.UnexpectedEnumCase();
+                case "{{ variant.name()|class_name(ci) }}": {
+                    intConverter.write({{ loop.index }}, into);
+                    break;
+                }
+            {%- endfor %}
+                default:{
+                    throw new UniffiInternalError.UnexpectedEnumCase();
+                }
```

Comment:
Changed all this because hermes object construction ends up with things like:

```ts
class SimpleError extends Error {}
class ErrorVariant extends SimpleError {}

const e = new ErrorVariant("foo");
if (e instanceof ErrorVariant) {
  console.log("What Spidermonkey says");
} else if (e instanceof Error) {
  console.log("What hermes says");
}
```